client/app/components/QueryControlDropdown.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Dropdown from 'antd/lib/dropdown';
+import Menu from 'antd/lib/menu';
+import Button from 'antd/lib/button';
+import Icon from 'antd/lib/icon';
+import { react2angular } from 'react2angular';
+
+
+export function QueryControlDropdown(props) {
+  const menu = (
+    <Menu>
+      {!props.query.isNew() && (!props.query.is_draft || !props.query.is_archived) && (
+        <Menu.Item>
+          <a target="_self" onClick={() => props.openAddToDashboardForm(props.selectedTab)}>
+            <Icon type="plus-circle" theme="filled" /> Add to Dashboard
+          </a>
+        </Menu.Item>
+      )}
+      {!props.query.isNew() && (
+        <Menu.Item>
+          <a onClick={() => props.showEmbedDialog(props.query, props.selectedTab)} data-test="ShowEmbedDialogButton">
+            <Icon type="share-alt" /> Embed elsewhere
+          </a>
+        </Menu.Item>
+      )}
+      {(props.queryExecuting || !props.queryResultData) && (
+        <Menu.Item>
+          <a query-result-link="true" target="_self">
+            <Icon type="file" /> Download as CSV File
+          </a>
+        </Menu.Item>
+      )}
+      {(props.queryExecuting || !props.queryResultData) && (
+        <Menu.Item>
+          <a query-result-link="true" file-type="xlsx" target="_self">
+            <Icon type="file-excel" /> Download as Excel File
+          </a>
+        </Menu.Item>
+      )}
+    </Menu>
+  );
+
+  return (
+    <Dropdown trigger={['click']} overlay={menu}>
+      <Button><Icon type="ellipsis" rotate={90} /></Button>
```

Comment:
The 15px side padding of the button, make it look too wide:

![Screen Shot 2019-04-20 at 8 20 07](https://user-images.githubusercontent.com/486954/56453097-24096700-6345-11e9-94f9-a2a4f0a21bdb.png)


This should make it a square w/o extra markup:
```less
// ant.less
.@{btn-prefix-cls} .@{icon-prefix-cls}-ellipsis {
  margin: 0 -7px;
}
```
![Screen Shot 2019-04-20 at 8 19 53](https://user-images.githubusercontent.com/486954/56453098-2e2b6580-6345-11e9-8dab-5a52f338f736.png)

Reply:
Good call.